### PR TITLE
Rule S2612 [VB.Net]: Setting loose file permissions is security-sensitive

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.VisualBasic/Rules/LooseFilePermissions.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                var diagnostic = invocationLocation.GetLineSpan().StartLinePosition.Line == secondaryLocation.GetLineSpan().StartLinePosition.Line
+                    ? Diagnostic.Create(Rule, invocationLocation)
+                    : Diagnostic.Create(Rule, invocationLocation, additionalLocations: new[] {secondaryLocation});
+
+                context.ReportDiagnosticWhenActive(diagnostic);
+            }
+        }
+
+        private static bool IsFileAccessPermissions(SimpleNameSyntax identifierNameSyntax, SemanticModel semanticModel) =>
+            LooseFilePermissionsConfig.WeakFileAccessPermissions.Contains(identifierNameSyntax.Identifier.Text)
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.VisualBasic/Rules/LooseFilePermissions.cs:73]: Maybe identifierNameSyntax.Identifier could go into the SyntaxFacade class and if that is possible than the whole method could go to the base class.

File: analyzers/src/SonarAnalyzer.VisualBasic/Rules/LooseFilePermissions.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+{
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+    [Rule(DiagnosticId)]
+    public sealed class LooseFilePermissions : LooseFilePermissionsBase<SyntaxKind>
+    {
+        protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+
+        public LooseFilePermissions() : this(AnalyzerConfiguration.Hotspot) { }
+
+        internal LooseFilePermissions(IAnalyzerConfiguration configuration) : base(configuration, RspecStrings.ResourceManager)
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.VisualBasic/Rules/LooseFilePermissions.cs:42]: styling: can be single line with { }

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM, if you want you can move that method to the base class.

File: analyzers/src/SonarAnalyzer.VisualBasic/Rules/LooseFilePermissions.cs (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.VisualBasic/Rules/LooseFilePermissions.cs:73]: I'm not sure about this. In order to move `identifierNameSyntax.Identifier` there, we would need to use the base class `SyntaxNode` and do a cast in derived types. With this we will lose the type safety check which is now done at the compile time.

Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.VisualBasic/Rules/LooseFilePermissions.cs]: Thanks for the tip with `Language.Syntax.NodeIdentifier(syntaxNode)`. Looks better now 👍

**Pull request merged**